beacon_node/network/src/sync/block_lookups/single_block_lookup.rs

```diff
@@ -272,19 +272,16 @@ impl<L: Lookup, T: BeaconChainTypes> SingleBlockLookup<L, T> {
     pub(crate) fn missing_blob_ids(&self) -> MissingBlobs {
         let block_root = self.block_root();
         if let Some(components) = self.child_components.as_ref() {
-            self.da_checker.get_missing_blob_ids(block_root, components)
-        } else {
-            let Some(processing_availability_view) =
-                self.da_checker.get_processing_components(block_root)
-            else {
-                return MissingBlobs::new_without_block(block_root, self.da_checker.is_deneb());
-            };
             self.da_checker
-                .get_missing_blob_ids(block_root, &processing_availability_view)
+                .get_missing_blob_ids(block_root, Some(components))
+        } else {
+            // TODO: the da_checker should look into the availability cache, not only the processing
+            // cache.
+            self.da_checker.get_missing_blob_ids_with(block_root)
```

Comment:
@realbigsean on unstable single lookup only checks the processing availability view to know what to download, ignoring the availability cache. Was this by design or a typo? Blobs are more likely to be in the latter if the intent is to not re-download pending components

Reply:
There was just the availability cache originally. But the availability cache is only populated after the execution payload is verified and its contents are removed before the blocks/blobs are saved to disk.   

We were seeing examples of us requesting blocks/blobs that were in the middle of execution payload verification for example, because they hadn't hit the availability cache yet. So we'd send a single lookup and get a duplicate download. The processing cache is meant to track not only what is in the availability cache, but also what is going to enter it soon, and what has just left it and is in the process of being written to storage. So the information they store is somewhat redundant, but the processing cache is meant to be a superset of what's in the availability cache.

Eviction from the processing cache occurs when the availability cache does *not* return `AvailabilityProcessingStatus::MissingComponents`. `MissingComponents` implies something was cached in the availability cache.

```
    fn remove_notified(
        &self,
        block_root: &Hash256,
        r: Result<AvailabilityProcessingStatus, BlockError<T::EthSpec>>,
    ) -> Result<AvailabilityProcessingStatus, BlockError<T::EthSpec>> {
        let has_missing_components =
            matches!(r, Ok(AvailabilityProcessingStatus::MissingComponents(_, _)));
        if !has_missing_components {
            self.data_availability_checker.remove_notified(block_root);
        }
        r
    }
```

Reply:
To prevent double downloads, maybe the de-duplication gadget should sit even more higher up? i.e. If you have a block in the beacon processor queue incoming from gossip you still should not attempt a download.

Reply:
> To prevent double downloads, maybe the de-duplication gadget should sit even more higher up? i.e. If you have a block in the beacon processor queue incoming from gossip you still should not attempt a download.

sounds like a good idea to me